OtripleS.Portal.Web.Tests.Unit/Services/StudentViews/StudentViewServiceTests.Logic.RetrieveAll.cs

```diff
@@ -0,0 +1,83 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE AS LONG AS SOFTWARE FUNDS ARE DONATED TO THE POOR
+// ---------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Moq;
+using OtripleS.Portal.Web.Models.Students;
+using OtripleS.Portal.Web.Models.StudentViews;
+using Xunit;
+
+namespace OtripleS.Portal.Web.Tests.Unit.Services.StudentViews
+{
+    public partial class StudentViewServiceTests
+    {
+        [Fact]
+        public async Task ShouldRetrieveAllStudentsViewAsync()
+        {
+            // given
+            var randomUserId = Guid.NewGuid();
+            DateTimeOffset randomDateTime = GetRandomDate();
+
+            dynamic dynamicStudentProperties =
+                CreateRandomStudentViewProperties(
+                    auditDates: randomDateTime,
+                    auditIds: randomUserId);
+
+            var student = new Student
+            {
```

Comment:
You need to generate a *list* of dynamic properties so we can test the *list* aspect of our functionality

Reply:
Then need to assign those list of dynamic properties to Student and StudentView, need to iterate over the list? Is this the approach that you said?

Reply:
You should create something like:

```csharp
List<dynamic> studentViewPropertiesCollection = CreateRandomStudentViewCollections();

List<Student> randomStudents = studentViewPropertiesCollection.Select(property =>
  new Student
  {
      // do your mapping here from property to student
  }).ToList();

List<Student> retrievedStudents = randomStudents;

List<StudentViews> randomStudentViews = studentViewPropertiesCollection.Select(property =>
  new StudentView
  {
      // do your mapping here from property to student
  }).ToList();

List<StudentView> expectedStudentViews = randomStudentViews;
```

Something to that effect.

Reply:
Thanks.